### PR TITLE
Improve new application password login flow

### PR DESCRIPTION
**Pull request by crazytonyli**
There are two main changes in this PR:
1. The core of application-password authentication has been moved to wordpress-rs in https://github.com/Automattic/wordpress-rs/pull/250.
2. Add a logo to the new site address view, as suggested by @kean in https://github.com/wordpress-mobile/WordPress-iOS/pull/23484#pullrequestreview-2228512255.

## Screenshots

Please note: I didn't change this view to a modal, because that'll turn this login flow into modals on top of modals: site picker -> typing address -> web view to authenticate.

| Enter | Typing | Start login |
| - | - | - |
| Screenshot 2024-08-13 at 1 18 57 PM | Screenshot 2024-08-13 at 1 19 09 PM | Screenshot 2024-08-13 at 1 19 12 PM |

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Verify login flow continues to work

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>Modules/Package.swift</code> was changed without updating its corresponding <code>Package.resolved</code>. Please resolve the Swift packages in Xcode.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/Login/LoginWithUrlView.swift [hunk 1]
```diff
@@ -47,31 +60,45 @@ struct LoginWithUrlView: View {
...
-                    )
-                    .stroke(.primary, lineWidth: 2)
-                )
-            })
-            .tint(.primary)
-        }.padding()
+                .disabled(isContinueButtonDisabled)
+
+            if isContinueButtonDisabled {
+                button.buttonStyle(.bordered)
```
Comment (kean) [inline WordPress/Classes/Login/LoginWithUrlView.swift]: This button doesn't look quite right. I suggest using the existing `DSButton` for now until there is a better replacement.

```swift
DSButton(title: SharedStrings.Button.continue, style: DSButtonStyle(emphasis: .primary, size: .large), isLoading: isLoading) {
    // TODO: handle action
}
.disabled(isContinueButtonDisabled)
```

 A better replacement should be implemented using native `ButtonStyle` API so you could use it as a modifier. I added the following type recently but it doesn't support the full-width presentation yet:

```swift
public struct PrimaryButtonStyle: ButtonStyle {
```

File: WordPress/Classes/Login/LoginWithUrlView.swift [hunk 2]
```diff
@@ -47,31 +60,45 @@ struct LoginWithUrlView: View {
...
+                button.buttonStyle(.borderedProminent)
+            }
+        }
+        .padding()
+        .navigationTitle(Self.title)
+        .navigationBarTitleDisplayMode(.inline)
     }
 
-    func startLogin() {
+    private func buttonContent() -> some View {
```
Comment (kean) [inline WordPress/Classes/Login/LoginWithUrlView.swift]: Unused code.

File: WordPress/Classes/Login/LoginWithUrlView.swift [hunk 3]
```diff
@@ -16,25 +16,38 @@ struct LoginWithUrlView: View {
...
+                    .frame(width: 60, height: 60)
+                    .padding(24)
+                Text(Self.enterSiteAddress)
+            }
 
             TextField(text: $urlField) {
                 Text("example.com")
             }
             .padding(.vertical)
             .overlay(Divider(), alignment: .bottom)
```
Comment (kean) [inline WordPress/Classes/Login/LoginWithUrlView.swift]: The underline is too far from the view and looks like a separate divider. It was the case in the previous implementation also.

File: WordPress/Classes/Login/LoginWithUrlView.swift [hunk 4]
```diff
@@ -16,25 +16,38 @@ struct LoginWithUrlView: View {
...
-        self.client = LoginClient(session: URLSession(configuration: .ephemeral))
+        self.client = client
         self.anchor = anchor
         self.loginCompleted = loginCompleted
     }
 
     var body: some View {
         VStack(alignment: .leading) {
-            Text("Enter the address of the WordPress site you'd like to connect.").padding(.vertical)
+            VStack(alignment: .center, spacing: 16) {
```
Comment (kean) [inline WordPress/Classes/Login/LoginWithUrlView.swift:36]: (nit) The `body` is getting too long. I would suggest extracting something.

File: WordPress/Classes/Login/LoginWithUrlView.swift [hunk 5]
```diff
@@ -89,26 +116,28 @@ struct LoginWithUrlView: View {
     }
 }
 
+extension LoginWithUrlView {
```
Comment (kean) [inline WordPress/Classes/Login/LoginWithUrlView.swift]: Please make it `private`.

I think I like this approach more than `Strings`, but just noting that it's non-standard.

**Review by kean — CHANGES_REQUESTED** [review]
Smoke test passed. I left a couple of comments.

**Comment by crazytonyli** [comment]
@kean Thanks for the review. I have addressed your comments in the commits above ⬆️ .

**Comment by kean** [comment]
The changes look good. I would appreciate if you could hold off merging until https://github.com/wordpress-mobile/WordPress-iOS/pull/23498 is merged.

**Review by kean — APPROVED** [review]

**Review by jkmassel — APPROVED** [review]
LGTM – I think we can further improve on this but it's a great start!!

**Pull request merged**